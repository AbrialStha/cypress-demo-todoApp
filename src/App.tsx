import React, { Component } from 'react';
import './index.css'
import TextField from './Components/TextField';
import { View, Todos, Todo } from './Components/view';
import { getTodos, addTodo, deleteTodo } from './utils/Request'

class App extends Component<{}, any> {
  constructor(props: Todos) {
    super(props)
    this.state = {
      "todos": []
    }
  }

  componentDidMount() {
    getTodos().then((resp: Todos) => {
      this.setState({
        todos: resp
      })
    })
  }

  addTodo: (todo: string) => void = (todo: string) => {
    let todos = this.state.todos
    let new_id;
    if (todos.length > 0)
      new_id = todos[todos.length - 1].id + 1
    else
      new_id = 0
    let new_todo = {
      "id": new_id,
      "title": todo,
      "completed": false
    }

    addTodo(new_todo).then((resp: Todo) => {
      this.setState({
        todos: [...todos, resp]
      })
    })
  }

  deleteTodo: (id: number) => void = (id: number) => {
    let new_todos = this.state.todos.filter((todo: Todo) => todo.id != id)
    deleteTodo(id).then((resp: Todos) => {
      this.setState({
        todos: new_todos
      })
    })
  }

  toogleTodo: (id: number) => void = (id: number) => {
    let new_todos = this.state.todos.map((todo: Todo) => {
      if (todo.id == id) {
        todo.completed = !todo.completed
        return todo
      }
    })
    this.setState({
      todo: [...new_todos]
    })
  }

  render() {
    return <div className="main container">
      <h1>Simple Todo App</h1>
      <TextField addTodo={this.addTodo} />
      <br />
      <View todos={this.state.todos} deleteTodo={this.deleteTodo} toogleTodo={this.toogleTodo} />
    </div>;
  }
}

export default App;
