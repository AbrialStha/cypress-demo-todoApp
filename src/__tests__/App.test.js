import React from "react";
import ReactDom from "react-dom";
import App from "../App";

import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<App />, div);
  ReactDom.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

