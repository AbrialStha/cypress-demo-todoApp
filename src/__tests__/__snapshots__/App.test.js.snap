// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="main container"
>
  <h1>
    Simple Todo App
  </h1>
  <div
    className="input-group mb-3"
  >
    <input
      className="input-group-text"
      onChange={[Function]}
      style={
        Object {
          "width": "60%",
        }
      }
      type="text"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
  <br />
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td
          style={
            Object {
              "width": "50px",
            }
          }
        >
          <input
            defaultChecked={false}
            onClick={[Function]}
            type="checkbox"
          />
        </td>
        <td
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Do something
        </td>
        <td
          style={
            Object {
              "width": "120px",
            }
          }
        >
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "width": "50px",
            }
          }
        >
          <input
            defaultChecked={true}
            onClick={[Function]}
            type="checkbox"
          />
        </td>
        <td
          style={
            Object {
              "textDecoration": "line-through",
            }
          }
        >
          <span>
            <span
              className="badge badge-pill badge-success"
            >
              completed
            </span>
             
          </span>
          something Done
        </td>
        <td
          style={
            Object {
              "width": "120px",
            }
          }
        >
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`matches snapshot without the two todos 1`] = `
<div
  className="main container"
>
  <h1>
    Simple Todo App
  </h1>
  <div
    className="input-group mb-3"
  >
    <input
      className="input-group-text"
      onChange={[Function]}
      style={
        Object {
          "width": "60%",
        }
      }
      type="text"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
  <br />
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td
          style={
            Object {
              "width": "50px",
            }
          }
        >
          <input
            defaultChecked={false}
            onClick={[Function]}
            type="checkbox"
          />
        </td>
        <td
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Do something
        </td>
        <td
          style={
            Object {
              "width": "120px",
            }
          }
        >
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "width": "50px",
            }
          }
        >
          <input
            defaultChecked={true}
            onClick={[Function]}
            type="checkbox"
          />
        </td>
        <td
          style={
            Object {
              "textDecoration": "line-through",
            }
          }
        >
          <span>
            <span
              className="badge badge-pill badge-success"
            >
              completed
            </span>
             
          </span>
          something Done
        </td>
        <td
          style={
            Object {
              "width": "120px",
            }
          }
        >
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
